dialogue-jmh/build.gradle

```diff
@@ -0,0 +1,52 @@
+versionsLock {
+    testProject()
+}
+
+dependencies {
+    implementation project(':dialogue-clients')
+    implementation project(':dialogue-core')
+    implementation project(':dialogue-example:dialogue-example-dialogue')
+    implementation project(':dialogue-target')
+    implementation project(':dialogue-test-common')
+    implementation 'com.github.ben-manes.caffeine:caffeine'
+    implementation 'com.google.guava:guava'
+    implementation 'com.palantir.conjure.java.runtime:client-config'
+    implementation 'com.palantir.safe-logging:preconditions'
+    implementation 'com.palantir.safe-logging:safe-logging'
+    implementation 'com.palantir.safethreadlocalrandom:safe-thread-local-random'
+    implementation 'com.palantir.tracing:tracing'
+    implementation 'com.palantir.tritium:tritium-registry'
+    implementation 'com.palantir.tritium:tritium-metrics'
+    implementation 'io.undertow:undertow-core'
+    implementation 'org.openjdk.jmh:jmh-core'
+
+    runtimeOnly 'org.apache.logging.log4j:log4j-slf4j-impl'
+    runtimeOnly 'org.apache.logging.log4j:log4j-core'
+
+    annotationProcessor 'org.openjdk.jmh:jmh-generator-annprocess'
+    annotationProcessor 'org.immutables:value'
+    compileOnly 'org.immutables:value::annotations'
+}
+
+
+def benchmarkFile = "${buildDir}/benchmark_results.out"
+
+task jmh(type: JavaExec) {
+    main = 'org.openjdk.jmh.Main'
+    classpath = sourceSets.main.runtimeClasspath
+    maxHeapSize = '1024m'
+
+    // e.g. ./gradlew jmv -PjmhOptions='-rf json'
+    if (project.hasProperty('jmhOptions')) {
+        args(project.jmhOptions.split(' '))
+    }
+    doFirst {
+        standardOutput new org.apache.tools.ant.util.TeeOutputStream(
+                new FileOutputStream(benchmarkFile), System.out)
```

Comment:
Unless we're using the output I'd prefer to leave this out, iirc it interacts poorly with ctrl+c and the benchmarks continue running.